pycytominer/cyto_utils/cells.py

```diff
@@ -430,46 +435,210 @@ def aggregate_compartment(
-            fields_count_df = self.image_df.loc[
-                :, list(np.union1d(self.strata, self.fields_of_view_feature))
-            ]
-            fields_count_df = (
-                fields_count_df.groupby(self.strata)[self.fields_of_view_feature]
-                .count()
-                .reset_index()
-                .rename(
-                    columns={f"{self.fields_of_view_feature}": f"Metadata_Site_Count"}
+            if compute_counts and self.fields_of_view_feature not in self.strata:
+                fields_count_df = self.image_df.loc[
+                    :, list(np.union1d(self.strata, self.fields_of_view_feature))
+                ]
+                fields_count_df = (
+                    fields_count_df.groupby(self.strata)[self.fields_of_view_feature]
+                    .count()
+                    .reset_index()
+                    .rename(
+                        columns={
+                            f"{self.fields_of_view_feature}": f"Metadata_Site_Count"
+                        }
+                    )
                 )
-            )
 
-            object_df = fields_count_df.merge(object_df, on=self.strata, how="right")
+                partial_object_df = fields_count_df.merge(
+                    partial_object_df,
+                    on=self.strata,
+                    how="right",
+                )
+
+            object_dfs.append(partial_object_df)
+
+        # Concatenate one or more aggregated dataframes row-wise into final output
+        object_df = pd.concat(object_dfs, axis=0).reset_index(drop=True)
 
         return object_df
 
+    def _compartment_df_generator(
+        self,
+        compartment,
+        n_strata=1,
+    ):
+        """A generator function that returns chunks of the entire compartment
+        table from disk.
+
+        We want to return dataframes with all compartment entries within unique
+        combinations of self.merge_cols when aggregated by self.strata
+
+        Parameters
+        ----------
+        compartment : str
+            Compartment to aggregate.
+        n_strata : int, default 1
+            Number of unique strata to pull from the database into working memory
+            at once.  A larger number does not lead to faster compute times
+
+        Yields
+        ------
+        image_df : pandas.core.frame.DataFrame
+            A generator whose __next__() call returns a chunk of the compartment
+            table, where rows comprising a unique aggregation stratum are not split
+            between chunks, and thus groupby aggregations are valid
+
+        """
+
+        assert (
+            n_strata > 0
+        ), "Number of strata to pull into memory at once (n_strata) must be > 0"
+
+        # Columns of the compartment table we will obtain
+        cols = "*"
+        compartment_row1 = pd.read_sql(
+            sql=f"select {cols} from {compartment} limit 1",
```

Comment:
what does `limit 1` mean? it will pull only the first row?

Reply:
Yes, that's right